src/aws-documentation-mcp-server/tests/test_aws_search_live.py

```diff
@@ -127,6 +137,94 @@ async def test_search_documentation_limit():
         print(f'Received {len(results_large)} results with limit=10')
 
 
+@pytest.mark.asyncio
+@pytest.mark.live
+async def test_search_documentation_with_product_type():
+    """Test search_documentation with an initial search then use product filter for a second search with same query."""
+    search_phrase = 'AWS Lambda'
+    ctx = MockContext()
+
+    with patch(
+        'awslabs.aws_documentation_mcp_server.server_aws.DEFAULT_USER_AGENT',
+        TEST_USER_AGENT,
+    ):
+        # First search without filters
+        first_response = await search_documentation(
+            ctx, search_phrase=search_phrase, limit=10, product_types=None, guide_types=None
+        )
+        first_results = first_response.searchResults
+        # Verify first search results
+        assert first_response is not None
+        assert first_results is not None
+
+        # Get first product filter facet
+        assert first_response.facets is not None
+        assert 'product_types' in first_response.facets
+        product_types = first_response.facets['product_types']
+        assert len(product_types) > 0
+
+        first_product_type = product_types[0]
+
+        # Second search with the product filter
+        second_response = await search_documentation(
+            ctx,
+            search_phrase=search_phrase,
+            limit=10,
+            product_types=[first_product_type],
+            guide_types=None,
+        )
+        second_results = second_response.searchResults
+
+        # Verify second search results
+        assert second_response is not None
+        assert second_results is not None
+        assert len(second_results) > 0
```

Comment:
Is there a verification from the SearchResponse object that can be made to confirm that `product_types` were used to filter? Same for `guide_types` in the next test.